Sources/NIO/SelectableEventLoop.swift

```diff
@@ -616,7 +620,12 @@ Further information:
     @inlinable
     public func makeSucceededVoidFuture() -> EventLoopFuture<Void> {
         guard self.inEventLoop, let voidFuture = self._succeededVoidFuture else {
-            return EventLoopFuture(eventLoop: self, value: (), file: "n/a", line: 0)
+            // We have to create the promise and complete it because otherwise we'll hit a loop in `makeSucceededFuture`. This is
+            // fairly dumb, but it's the only option we have. This one can only happen after the loop is shut down,
+            // which is an acceptable compromise.
+            let voidPromise = self.makePromise(of: Void.self)
+            voidPromise.succeed(())
+            return voidPromise.futureResult
         }
         return voidFuture
```

Comment:
Why, do we need to check, if we are on the EL, the method is called on?

Reply:
`_succeededVoidFuture` creates a reference cycle which can leak an event loop (it's stored on `self` and holds a reference to its owning `EventLoop`, which is `self`). As a result, it is `nil`'d when we exit the event loop. To avoid needing a lock to read it, we only use this fast path if we're on the loop, where we can safely read it synchronously.

Reply:
Ah! No the comment makes a little more sense, but I think we should update the code comment to talk about a reference cycle instead of a loop. Also this seems to be out of date:

> This one can only happen after the loop is shut down, which is an acceptable compromise.

Reply:
There is a reference to the loop in the assignment to the variable in `init`.